app/controllers/admin/schedules_controller.rb

```diff
@@ -7,10 +7,11 @@ class SchedulesController < Admin::BaseController
     load_resource :venue, through: :conference, singleton: true
 
     skip_before_filter :verify_authenticity_token, only: [:update]
-    layout 'schedule'
 
     def show
-      authorize! :update, @program.events.new
+      event = @program.events.new
+      authorize! :update, event
+      event.destroy
```

Comment:
Because then I'll save that new event when doing `@program.save!`
But if I move that to the create action as you suggested then I can have just  `authorize! :update, @program.events.new`